### ISSUE TITLE
Performance of `std` vs. `liftstd`

### ISSUE THREAD

**Issue by zickgraf** [Singular/Singular#925]
I want to compute the transformation matrix from a given module over a super-commutative ring to a standard basis of this module. My current approach is to use `liftstd` (see code below). I expect the runtime of `liftstd` to be similar to the runtime of `std`, because in the algorithms I know, `liftstd` is "`std` + some bookkeeping". This is correct in some cases, but in other cases, `std` finishes in one second, whereas `liftstd` is still running after some minutes (I did not check if it terminates eventually).

Is this behaviour expected or should this be considered as a bug?

If this is a bug, I will provide a concrete example, of course, but since my example is large, I do not want to clutter this issue right now.

```
LIB "nctools.lib";

ring R = (0),(e0,e1,e2),dp;
def S = superCommutative(1);
setring S;

matrix M[702][216] = ...; // large matrix

matrix T;
module s;
std(module(M)); // terminates after one second
liftstd(module(M),T,s,"std"); // still running after some minutes
```

**Comment by zickgraf** [comment]
I have investigated this a bit more: the difference in performance also occurs for commutative rings. In one example, `std` finished in less than a second but `liftstd` takes nearly three minutes.

**Comment by hannes14** [comment]
solved with b88827821a3ed7373494ab9c988f91d03cfc2f68
